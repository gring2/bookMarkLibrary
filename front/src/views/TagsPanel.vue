<template>
    <aside class="side">
      <ul v-for="tag in tags">
        <TagLink :tag="tag"></TagLink>
      </ul>
    </aside>
</template>

<script lang="ts">
import { Component , Vue } from 'vue-property-decorator'
import TagLink from '@/components/TagLink.vue'
import Tag from '@/vo/Tag'
import {bookmarkMod} from '@/stores/modules/bookmark'


@Component({
  components: {
    TagLink,
  },
})
export default class TagsPanel extends Vue {
  private tags?: Tag[] = []
  private created() {
    this.tags = bookmarkMod.tags
  }
}
</script>

<style scoped lang="scss">
  aside{
    width: 25%
  }
</style>
