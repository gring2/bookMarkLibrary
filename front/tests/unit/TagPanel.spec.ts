import { mount, shallowMount } from '@vue/test-utils';
import TagsPanel from '@/views/TagsPanel.vue';

describe('Tag Penel Test', () => {
  it('render Tag component', () => {
    const wrapper = mount(TagsPanel, {
      data: () => {
        return {
          tags: ['test tag1', 'test tag2']
        }
      }
    });
    expect(wrapper.text().indexOf('test tag1')).toBeTruthy()
    expect(wrapper.text().indexOf('test tag2')).toBeTruthy()
  })
})