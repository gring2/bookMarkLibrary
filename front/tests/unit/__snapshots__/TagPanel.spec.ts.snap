// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagPenel Test render TagPenel view 1`] = `<aside class="side"></aside>`;
